File: num8.hs
module Main where

main = do
    putStrLn "By Thanh Nguyen"
    print $ area ("circle", [2, 4, 5])
    print $ area ("rectangle", [2, 4, 5])
    print $ area ("triangle", [2, 4, 5])
    print $ area ("haha", [2, 4, 5])
    
area :: ([Char], [Integer]) -> Float
area a =
    if fst a == "circle" then circ $ snd a
    else if fst a == "rectangle" then rect $ snd a
    else if fst a == "triangle" then tria $ snd a
    else -1

circ :: [Integer] -> Float
circ (r:xs) = fromIntegral r * fromIntegral r * pi

rect :: [Integer] -> Float
rect (l:w:xs) = fromIntegral l * fromIntegral w

tria :: [Integer] -> Float
tria (b:h:xs) = fromIntegral b * fromIntegral h / 2

File: num8.fs
module Main

let area (shape, list) = 
    match (shape,list) with
    | ("circle", r::xs) -> (r * r) * (System.Math.PI)
    | ("rectangle", l::w::xs) -> l * w
    | ("triangle", b::h::xs) -> b * h / 2.0
    | _ -> -1.0

printf "%s" ("By Thanh Nguyen")
printf "%.2f" (area ("circle",[2.; 4.; 5.]))
printf "%.2f" (area ("rectangle",[2.; 4.; 5.]))
printf "%.2f" (area ("triangle",[2.; 4.; 5.]))
printf "%.2f" (area ("haha",[2.; 4.; 5.])) 